src/main/java/org/jboss/xavier/integrations/route/MainRouteBuilder.java

```diff
@@ -317,6 +330,19 @@ public String getUserNameFromRHIdentity(String x_rh_identity_base64) {
         return result;
     }
 
+    public String getUserAccountFromRHIdentity(String x_rh_identity_base64) {
```

Comment:
As you are basically duplicating the code, probably you could refactor this method and mix it with the getUserNameFromRHIdentity in order to get both fields. Currently it is parsing the Json encoded value 2 times without need.

Reply:
Refactored